[TC-9065] Update webhook connectors (#5)
tradecloud/tradecloud-docs-connectors: webhook-connector/order-events.md

---
description: Order Events overview.
---

# Order Events

When using the order webhook you may configure and receive below order events.

## Order is issued or updated by buyer

The buyer can issue new order lines or update existing issued lines.

> If confirmed delivery schedule or prices are updated, this will become a reopen request.

Reviewer: ```suggestion
If confirmed delivery schedule or prices are updated, this will become a [reopen request](#order-reopen-request-by-buyer).
```

Author: Updated

| OrderEvent                  | Webhook Configuration |
| --------------------------- |-----------------------|
| `OrderIssuedByBuyer`        | New order is issued by the buyer |
| `OrderLinesIssuedByBuyer`   | Additional order lines are added by the buyer |
| `OrderLinesReissuedByBuyer` | Updated order lines are reissued by the buyer |
| `OrderLinesUpdatedByBuyer`  | Order and line fields other than prices & delivery schedules are updated by the buyer |

## Order confirmed by supplier or buyer

> The supplier can either accept, reject or propose an alternative (see next section).

Reviewer: ```suggestion
The supplier can either accept, reject or [propose an alternative](#order-proposal-by-supplier).
```

Author: Updated

> * Accepted means that the supplier confirmed the order line(s) as requested by the buyer.

Reviewer: Nit: put the explanations in the table below.

Author: The text in the table are the literal texts in the webhook configuration you may choose.

Reviewer: In the other tables it's a rephrasing / explanation of what such events mean. imo it makes sense to move these bullets to the table for consistency as well.

* Confirmed by buyer is used when the buyer migrates already confirmed order lines to Tradecloud.

| OrderEvent                      | Webhook Configuration                     |
| ------------------------------- | ----------------------------------------- |
| `OrderLinesAcceptedBySupplier`  | Order lines are accepted by the supplier  |
| `OrderLinesRejectedBySupplier`  | Order lines are rejected by the supplier  |
| `OrderLinesConfirmedBySupplier` | Order lines are confirmed by the supplier |
| `OrderLinesConfirmedByBuyer`    | Order lines are confirmed by the buyer    |

## Order proposal by supplier

The supplier can propose an alternative delivery schedule or prices.

The buyer either approves or rejects the proposal.

| OrderEvent                            | Webhook Configuration                                        |
| --------------------------------------| ------------------------------------------------------------ |
| `OrderChangesProposedBySupplier`      | Order changes are proposed by the supplier                   |

> | `OrderChangesProposalApprovedByBuyer` | By supplier proposed order changes are approved by the buyer |
> | `OrderChangesProposalRejectedByBuyer` | By supplier proposed order changes are rejected by the buyer |

Reviewer: ```suggestion
| `OrderChangesProposalApprovedByBuyer` | By order changes that were proposed by the supplier are approved by the buyer |
| `OrderChangesProposalRejectedByBuyer` | By order changes that were proposed by the supplier are rejected by the buyer |
```

Author: The text in the table are the literal texts in the webhook configuration you may choose.

Reviewer: I don't think that really adds value. In other tables it's a rephrasing. In this table I think the rephrasing also helps with the understanding of it.

Reviewer: Or do you mean compared to the portal? Imo then the portal should be updated too ;)

## Order updated by supplier

The supplier can update existing order lines.

> If delivery schedule or prices are updated this will become either a proposal or reopen request.

Reviewer: ```suggestion
If delivery schedule or prices are updated this will become either a [proposal](#order-proposal-by-supplier) or [reopen request](#order-reopen-request-by-supplier).
```

Author: Updated

| OrderEvent                     | Webhook Configuration                        |
| ------------------------------ | -------------------------------------------- |
| `OrderLinesUpdatedBySupplier`  | Order and line fields other than prices & delivery schedules are updated by the supplier |
| `OrderLinesItemDetailsChanged` | Order lines item details are changed by the supplier | 

## Order reopen request by buyer

The buyer can request to reopen confirmed order lines.

The supplier can either approve or reject the reopen request.

| OrderEvent                                  | Webhook Configuration                        |
| ------------------------------------------- | -------------------------------------------- |

> | `OrderLinesReopenRequestedByBuyer`          | Order lines reopen is requested by buyer     |
> | `OrderLinesReopenRequestApprovedBySupplier` | Buyer reopen request is approved by supplier |
> | `OrderLinesReopenRequestRejectedBySupplier` | Buyer reopen request is rejected by supplier |

Reviewer: ```suggestion
| `OrderLinesReopenRequestedByBuyer`          | Order lines reopen is requested by the buyer     |
| `OrderLinesReopenRequestApprovedBySupplier` | Buyer reopen request is approved by the supplier |
| `OrderLinesReopenRequestRejectedBySupplier` | Buyer reopen request is rejected by the supplier |
```

Author: The text in the table are the literal texts in the webhook configuration you may choose.

## Order reopen request by supplier

The supplier can request to reopen confirmed order lines.

The buyer can either approve or reject the reopen request.

| OrderEvent                               | Webhook Configuration                        |
| ---------------------------------------- | -------------------------------------------- |

> | `OrderLinesReopenRequestedBySupplier`    | Order lines reopen is requested by supplier  |
> | `OrderLinesReopenRequestApprovedByBuyer` | Supplier reopen request is approved by buyer |
> | `OrderLinesReopenRequestRejectedByBuyer` | Supplier reopen request is rejected by buyer |

Reviewer: ```suggestion
| `OrderLinesReopenRequestedBySupplier`    | Order lines reopen is requested by the supplier  |
| `OrderLinesReopenRequestApprovedByBuyer` | Supplier reopen request is approved by the buyer |
| `OrderLinesReopenRequestRejectedByBuyer` | Supplier reopen request is rejected by the buyer |
```

Author: The text in the table are the literal texts in the webhook configuration you may choose.

Author: I will commit anyway, portal to be improved

## Order lines cancelled by buyer

> The buyer can cancel order lines without request.

Reviewer: ```suggestion
The buyer can cancel order lines that don't contain a request.
```

| `CompletedOrderLinesRevertedByBuyer` | Completed order lines are reverted by the buyer |

## Order lines logistics

Both buyer and supplier can update delivery schedule logistics fields `status`, `etd` and `eta`

| OrderEvent                                             | Webhook Configuration |
| ------------------------------------------------------ | --------------------- |
| `OrderLinesDeliveryScheduleLogisticsUpdatedByBuyer`    | Delivery schedule logistics fields are updated by the buyer    |
| `OrderLinesDeliveryScheduleLogisticsUpdatedBySupplier` | Delivery schedule logistics fields are updated by the supplier |

## Order contacts

Both buyer and supplier can update their order contact.

| OrderEvent                         | Webhook Configuration                                |
| ---------------------------------- | ---------------------------------------------------- |
| `OrderContactReassignedByBuyer`    | Buyer order contacts are reassigned in the portal    |
| `OrderContactReassignedBySupplier` | Supplier order contacts are reassigned in the portal |

## Maintenance

Buyer and supplier can resend order lines to their ERP system.

> Order lines can be synced from the tradecloud.nl to the the Tradecloud One platform.

Reviewer: ```suggestion
Order lines can be synced from the tradecloud legacy platform to the the Tradecloud One platform.
```
